modules/api/api.py

```diff
@@ -205,7 +205,7 @@ def __init__(self, app: FastAPI, queue_lock: Lock):
         self.router = APIRouter()
         self.app = app
         self.queue_lock = queue_lock
-        #api_middleware(self.app)  # XXX this will have to be fixed
+        #api_middleware(self.app)  # FIXME: (legacy) this will have to be fixed
```

Comment:
I can't decide if I should delete this method and its imports now? With the new version of gradio and fastapi, this method won't work anyway.
Or leave it as it is?

Reply:
I'd say leave it for now. If it is certain that it can't be restored then that's an excellent argument to remove, and add a comment in an appropriate place so say why. But no urgency to include that right now unless it's interfering somehow. I'm not an API user, so can't provide useful comments; but this updated PR is much better for reviewability and seems fine to me, so thanks for that. altoiddealer is an API user, so (without wanting to dump responsibility on them) I'll hold off on pressing the button for a short while, for their opinion.

Reply:
Ok. Waiting for @altoiddealer